### DISCUSSION TITLE
Am I the only one who feels like 4bit training should not be a monkeypatch?

### DISCUSSION THREAD

**Discussion by ReflectionRip** [oobabooga/text-generation-webui#2247] [General]
There are so many 4-bit quantized models out there, and many people can't LoRA the larger 8bit models. It is a pain to get the monkeypatch working, and even if you succeed the LoRA may still have problems.

I feel that 4bit should just be a standard supported training mode in the base oobabooga!

**Comment by valden80** [comment]
With this patch VRAM consuming still high for train process. On my 3060 with 12GB I manage to start train LORA only for 7b model, 13b always result OOM. From what I read in docs - it's close to 8bit requirements.

So, maybe this patch is not good enough solution to spend efforts on it for better integration. There is alternative coming, as I heard...

**Comment by Zhincore** [comment]
From what I understand the monkey-patch is a transitional workaround before 4bit LoRA is supported in some libraries upstream, which I think has been ready for some time but still not integrated for some reason
